src/subscription/confirmation-modal/ConfirmationModal.jsx

```diff
@@ -0,0 +1,86 @@
+import React, { useState } from 'react';
+import {
+  ModalDialog, ActionRow, Button, Hyperlink,
+} from '@edx/paragon';
+import { useSelector } from 'react-redux';
+import { ArrowForward } from '@edx/paragon/icons';
+import { getConfig } from '@edx/frontend-platform';
+import { useIntl, defineMessages, FormattedMessage } from '@edx/frontend-platform/i18n';
+import { detailsSelector } from '../data/details/selectors';
+
+const messages = defineMessages({
+  'subscription.confirmation.modal.heading': {
+    id: 'subscription.confirmation.modal.heading',
+    defaultMessage: 'Congratulations! Your 7-day free trial of {programTitle} has started.',
+    description: 'Subscription confirmation success heading.',
+  },
+  'subscription.confirmation.modal.body': {
+    id: 'subscription.confirmation.modal.body',
+    defaultMessage: "When your free trial ends, your subscription will begin, and we'll charge your payment method on file {price} per month plus any applicable taxes. This subscription will automatically renew every month unless you cancel from the {ordersAndSubscriptionLink} page.",
+    description: 'Subscription confirmation success message explaining monthly subscription plan.',
+  },
+  'subscription.confirmation.modal.body.orders.link': {
+    id: 'subscription.confirmation.modal.body.orders.link',
+    defaultMessage: 'Orders & Subscriptions',
+    description: 'Subscription Orders & Subscriptions link placeholder.',
+  },
+});
+
+/**
+ * ConfirmationModal
+ */
+export const ConfirmationModal = () => {
+  const [isOpen] = useState(false);
```

Comment:
my understanding is this will always be `true` and user won't be able to hide it. And we can conditionally render it according to the successful payment flow once that is done.